File: src/main/scala/universalelectricity/core/grid/node/NodeElectric.java
package universalelectricity.core.grid.node;

import net.minecraftforge.common.util.ForgeDirection;
import universalelectricity.api.core.grid.INodeProvider;
import universalelectricity.api.core.grid.electric.IElectricNode;

/**
 * Created by robert on 8/14/2014.
 */
public class NodeElectric extends NodeEnergy implements IElectricNode
{
    public NodeElectric(INodeProvider parent)
    {
        super(parent);
    }

    public NodeElectric(INodeProvider parent, double cap)
    {
        super(parent, cap);
    }

    public NodeElectric(INodeProvider parent, double cap, double tran)
    {
        super(parent, cap, tran);
    }

    public NodeElectric(INodeProvider parent, double cap, double in, double out)
    {
        super(parent, cap, in, out);
    }

    @Override
    public double getVoltage(ForgeDirection direction)
    {
        return 0;
    }
}

File: src/main/scala/resonant/lib/grid/electric/NodeElectric.scala
package resonant.lib.grid.electric

import net.minecraftforge.common.util.ForgeDirection
import resonant.api.tile.INodeProvider
import resonant.api.tile.node.{NodeType, IPressure, IFlow}
import resonant.lib.grid.branch.NodeBranchPart
import resonant.lib.grid.energy.EnergyStorage
import resonant.lib.prefab.TEnergyBuffer

import scala.beans.BeanProperty

/**
 * A node that runs on electricity
 * @author DarkCow
 */
class NodeElectric(parent: INodeProvider) extends NodeBranchPart(parent) with TEnergyBuffer with IFlow with IPressure
{
  @BeanProperty
  var resistance: Double = 0

  var buffer : EnergyStorage = null

  //Constructor
  buffer = new EnergyStorage()

  override def getEnergyStorage: EnergyStorage = buffer

  /** Current flowing through this node */
  def current : Double =
  {
    return getFlow(NodeType.ELECTRICITY, ForgeDirection.UNKNOWN)
  }

  override def getFlow(t: NodeType, side: ForgeDirection): Double =
  {
    if(t == NodeType.ELECTRICITY || t == NodeType.ENERGY)
    {

    }
    return 0
  }

  /** Voltage across this node */
  def voltage : Double =
  {
    return getPressure(NodeType.ELECTRICITY, ForgeDirection.UNKNOWN)
  }

  override def getPressure(t: NodeType, side: ForgeDirection): Double =
  {
    if(t == NodeType.ELECTRICITY || t == NodeType.ENERGY)
    {

    }
    return 0
  }
}